File: 0868-binary-gap/0868-binary-gap.py
class Solution:
    def binaryGap(self, n: int) -> int:
        curr = 0
        prev = -1
        result = 0
        while n>0:
            if(n&1) > 0:
                result = max(result,curr-prev) if prev!= -1 else result
                prev = curr
            curr+=1
            n >>= 1
        return result

File: 0868-binary-gap/0868-binary-gap.cpp
class Solution {
public:
    int binaryGap(int n) {
        int prev = -1;
        int curr = 0;
        int result = 0;
        while (n>0){
            if ((n&1) > 0){
                result = prev!=-1 ? max(result,curr-prev): result;
                prev = curr;
            }
            curr++;
            n>>=1;
        }
        return result;
    }
};